src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -2379,6 +2379,111 @@ paths:
                 $ref: '#/components/schemas/ErrorModel'
       x-codegen-request-body-name: upgrade
 
+##############################################################################
+## SEARCH API STANDARD ENDPOINTS
+##############################################################################
+  /api/repository/v1/search/{id}/index:
+    post:
+      tags:
+        - search
+        - snapshots
+        - repository
+      description: Create a new search index for a snapshot
+      operationId: createSearchIndex
+      parameters:
+        - name: id
+          in: path
+          description: A UUID to used to identify an object in the repository
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Index to create from SQL
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SearchIndexRequest'
+        required: true
+      responses:
+        200:
+          description: Return index result
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SearchIndexModel'
+        400:
+          description: Bad request - invalid id, badly formed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+        403:
+          description: No permission to see snapshot
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+        404:
+          description: Not found - snapshot id does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/search/query:
+    post:
+      tags:
+        - search
+        - snapshots
```

Comment:
Do we maybe want to keep this out of snapshots initially?

One thing I was wonder...does it make sense to have it in an 'experimental' tag?

Reply:
The API only appears under snapshots when the feature is enabled, but I also think it makes sense as it's "snapshot related", as indexing and querying are on snapshots. See my comment regarding having experimental endpoints here: https://github.com/DataBiosphere/jade-data-repo/pull/915#discussion_r631353914 . If we add it as a tag instead, I think we'd need to create a separate controller too, but I'm not sure.